### PR TITLE
[DCOS-51573][DCOS-51574] authn authz implementation

### PR DESCRIPTION
**Pull request by viivek46**
[DCOS-51573][DCOS-51574] authn authz implementation
## Release Notes 

With this Release internal authentication and internal authorization will be added to this framework.
 
Cassandra uses rolenames and passwords for internal authentication. Role-based authentication encompasses both users and roles to bring a number of useful features to authorization. Roles can represent either actual individual users or roles that those users have in administering and accessing the Cassandra cluster.

Design Document :- 
https://docs.google.com/document/d/1L2MR7czxXYcfHxlKLMQGNGVq5cj4L0wXQblIKD6cny8/edit?usp=sharing

### CODE REVIEW

File: frameworks/cassandra/universe/config.json [hunk 1]
```diff
@@ -107,8 +107,78 @@
...
+                            "name": {
+                                "type": "string",
+                                "description": "The name of the superuser",
+                                "default": "dcossuperuser"
+                            },
+                            "password_secret_path": {
+                                "type": "string",
+                                "description": "The path in the DC/OS Secret Store to retrieve the superuser password from" ,
+                                "default": "cassandra/password"
+			                      }
```
Comment (samvantran) [inline frameworks/cassandra/universe/config.json]: minor nit: formatting is off here

File: frameworks/cassandra/src/main/dist/cassandra.yaml [hunk 2]
```diff
@@ -111,8 +111,8 @@ snapshot_before_compaction: false
 auto_snapshot: {{CASSANDRA_AUTO_SNAPSHOT}}
 cross_node_timeout: false
 endpoint_snitch: GossipingPropertyFileSnitch
-roles_update_interval_in_ms: {{CASSANDRA_ROLES_UPDATE_INTERVAL_IN_MS}}
-permissions_update_interval_in_ms: {{CASSANDRA_PERMISSIONS_UPDATE_INTERVAL_IN_MS}}
```
Comment (akirillov) [inline frameworks/cassandra/src/main/dist/cassandra.yaml]: what is the reason for `CASSANDRA_` prefix being dropped for the new/updated parameters while is present for the rest of config options?

Comment (viivek46) [reply frameworks/cassandra/src/main/dist/cassandra.yaml]: as variables are moved from Cassandra tab to service.security under config.json, so i changed marathon variable name as well.

File: frameworks/cassandra/universe/config.json (hunk 1 above)
Comment (viivek46) [reply frameworks/cassandra/universe/config.json]: updated formatting

File: frameworks/cassandra/tests/test_auth.py [hunk 3]
```diff
@@ -49,6 +53,9 @@ def test_auth() -> None:
     config.verify_client_can_write_read_and_delete_with_auth(
         config.get_foldered_node_address(),
     )
+    tasks = sdk_tasks.get_service_tasks(config.SERVICE_NAME, "node-0")[0]
+    _, stdout, stderr = sdk_cmd.run_cli("task exec {} bash -c 'export JAVA_HOME=$(ls -d $MESOS_SANDBOX/jdk*/jre/) ;  export PATH=$MESOS_SANDBOX/python-dist/bin:$PATH ; export PATH=$(ls -d $MESOS_SANDBOX/apache-cassandra-*/bin):$PATH ; cqlsh -u dcossuperuser -p wrongpassword -e \"SHOW VERSION\" node-0-server.$FRAMEWORK_HOST $CASSANDRA_NATIVE_TRANSPORT_PORT' ".format(tasks.id))
+    assert "Provided username dcossuperuser and/or password are incorrect" in stderr
```
Comment (samvantran) [inline frameworks/cassandra/tests/test_auth.py]: Can you separate this portion out into its own test?
eg
```
def test_auth() -> None:
  ..

def test_unauthorized_users() -> None:
  ...
```
so each test's intent is a little more clear

**Pull request merged**